core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java

```diff
@@ -6019,6 +6019,87 @@ private HepProgram getTransitiveProgram() {
         .check();
   }
 
```

Comment:
Can we add some queries where the simplification happens having also some filters?

Reply:
Add testProjectJoinRemove17 and testProjectJoinRemove18 case.